pkg/api/types.go

```diff
@@ -669,6 +671,11 @@ type V20180331ARMManagedContainerService struct {
 	*v20180331.ManagedCluster
 }
 
+// CustomCloudProfile Represents custom cloud profile
+type CustomCloudProfile struct {
```

Comment:
Instead of adding a new sub-property `CustomCloudProfile` to `Properties`, have we considered adding an additional `AzureEnvironmentSpecConfig` for Azure Stack? Basically, how it is currently implemented:

- you specify which "cloud" you are in by declaring at string value for `Location`, which is a property of `ContainerService`.
- we have constructors which then return an instance of your particular `AzureEnvironmentSpecConfig`.

See how `AzureChinaCloudSpec` does this for mooncake scenarios.

I think I'd prefer to expand the `AzureEnvironmentSpecConfig` type definition, rather than add a new one to `Properties`.

Any thoughts here @CecileRobertMichon, @tariq1890, @mboersma?

Reply:
It is good idea to extend the existing object.
I am planning the following changes later to support ADFS authentication, customized AzureEnvironmentSpecConfig , I will extend customeCloudProfile to the following formart:
"customeCloudProfile":{
          "environment":{},
           "azureEnvironmentSpecConfig"{}
           "identitySystem":"ADFS",
           "AuthMethod": "Certificate"
}

if we extend the existing AzureEnvironmentSpecConfig, then I will need add the following fields
          "environment":{},
        "identitySystem":"ADFS",
          "AuthMethod": "Certificate"

@jackfrancis , please let me know your thoughts.

Reply:
and there will be a new sub-property `AzureEnvironmentSpecConfig` to `Properties` , since the configuration needs to be in API model.

Reply:
Understood, thanks for clarifying